addons/account/models/account_account.py

```diff
@@ -864,6 +1018,603 @@ def get_import_templates(self):
+        self.env.cr.execute(SQL(
+            """
+             UPDATE ir_property AS _ip1
+                SET res_id = %(updated_value)s
+              WHERE res_id IN %(values_to_update)s
+                AND id IN (
+                     SELECT DISTINCT ON (fields_id)
+                            id
+                       FROM ir_property
+                      WHERE res_id IN %(values_to_update)s
+                   ORDER BY fields_id, res_id
+                    )
+                AND NOT EXISTS (
+                     SELECT
+                       FROM ir_property AS _ip2
+                      WHERE _ip2.res_id = %(updated_value)s
+                        AND _ip2.fields_id = _ip1.fields_id
+                        AND _ip2.company_id = _ip1.company_id
+                    )""",
+            updated_value=f'account.account,{account_to_merge_into.id}',
+            values_to_update=tuple(f'account.account,{account_id}' for account_id in accounts_to_remove.ids),
+        ))
+
+        # Then delete any entries that were not updated by the query above
+        self.env.cr.execute(SQL(
+            """
+             DELETE FROM ir_property
+              WHERE res_id IN %(values_to_delete)s
+            """,
+            values_to_delete=tuple(f'account.account,{account_id}' for account_id in accounts_to_remove.ids),
+        ))
+
+        # Step 4: Remove merged accounts
+        self.env['account.account'].invalidate_model()
+        self.env.cr.execute(SQL(
+            """
+             DELETE FROM account_account
+              WHERE id IN %(account_ids_to_delete)s
+            """,
+            account_ids_to_delete=tuple(accounts_to_remove.ids),
+        ))
+
+        # Clear ir.model.data ormcache
+        self.env.registry.clear_cache()
+
+        # Step 5: Write company_ids and codes on the account
+        for company, code in code_by_company.items():
+            account_to_merge_into.with_company(company).code = code
+
+        account_to_merge_into.company_ids = company_ids_to_write
+
+    def action_unmerge(self):
+        """ Split the account `self` into several accounts, one per company.
+        The original account's codes are assigned respectively to the account created in each company.
+
+        From an accounting perspective, this does not change anything to the journal items, since their
+        account codes will remain unchanged. """
+
+        self._check_action_unmerge_possible()
+
+        self._action_unmerge_get_user_confirmation()
+
+        self._action_unmerge()
+
+        return {
+            'type': 'ir.actions.client',
+            'tag': 'display_notification',
+            'params': {
+                'type': 'success',
+                'sticky': False,
+                'message': _("Accounts successfully unmerged!"),
+                'next': {'type': 'ir.actions.act_window_close'},
+            }
+        }
+
+    def _check_action_unmerge_possible(self):
```

Comment:
Can you unmerge if it was locked?
Can't we just use the duplicate feature to unmerge?